File: BibliotecaPublica_TT/src/main/java/com/ReWo/BibliotecaPublica_TT/service/UsuarioService.java
package com.ReWo.BibliotecaPublica_TT.service;

import com.ReWo.BibliotecaPublica_TT.entity.Usuario;

import java.util.List;

public interface UsuarioService
{
    List<Usuario> listarTodos();

    Usuario buscarPorId(Long idUsuario);

    Usuario buscarPorEmail(String email);

    Usuario guardar(Usuario usuario);

    Usuario actualizar(Long idUsuario, Usuario usuarioActualizado);

    void eliminar(Long idUsuario);
}

File: biblioteca-frontend/src/services/usuarioService.js
import api from './api';

export const usuarioService = {
  listar: async () => (await api.get('/api/usuarios')).data,
  buscarPorId: async (id) => (await api.get(`/api/usuarios/${id}`)).data,
  buscarPorEmail: async (email) => (await api.get('/api/usuarios/email', { params: { email } })).data,
  crear: async (payload) => (await api.post('/api/usuarios', payload)).data,
  actualizar: async (id, payload) => (await api.put(`/api/usuarios/${id}`, payload)).data,
  eliminar: async (id) => (await api.delete(`/api/usuarios/${id}`)).data
};
